src/com/ubcsolar/sim/SimController.java

```diff
@@ -115,6 +115,110 @@ public void runSimulation(int laps, long startTime)
 
 		this.mySession.sendNotification(new NewSimulationReportNotification(toSend));
 	}
+	
+	public void runSimulationWithManualSpeeds(int laps, long startTime, Map<GeoCoord, Map<Integer, Double>> manualSpeedProfile)
+			throws NoForecastReportException, NoLoadedRouteException, NoLocationReportedException,
+			NoCarStatusException {
+		if (laps <= 0) {
+			throw new IllegalArgumentException("Number of Laps too low, must go at least 1 lap");
+		}
+		// Compile all the information we need.
+		LocationReport lastReported = this.mySession.getMapController().getLastReportedLocation();
+		if (lastReported == null) {
+			throw new NoLocationReportedException();
+		}
+		Route routeToTraverse = this.mySession.getMapController().getAllPoints();
+
+		if (routeToTraverse == null) {
+			throw new NoLoadedRouteException();
+		}
+		TelemDataPacket lastCarReported = this.mySession.getMyCarController().getLastTelemDataPacket();
+		if (lastCarReported == null) {
+			throw new NoCarStatusException();
+		}
+
+		SimResult result = null;
+		List<SimFrame> resultFrames = new ArrayList<SimFrame>(); //list of sim frames
+		//idea is to use runSimV2 for every part of the route where the car travels at constant speed until the speed change
+		//the point with the different speed would be the last point, and the point after would be the start of the next simulation
+		GeoCoord start = routeToTraverse.getTrailMarkers().get(1); //first point of the simulation (need to ask why its 1, running point 0 to 1 is always a fail)
```

Comment:
this is something we should deal with eventually, but not necessarily right now. I know there are some issues related to the sim if you give the first point 0 speed (right now we default to giving the speed as 1 km/h for the first point I think). There's probably some divide by 0 error somewhere that we should deal with, but it's not a high priority.